src/solvers/smt2/smt2_conv.cpp

```diff
@@ -4247,6 +4247,11 @@ void smt2_convt::set_to(const exprt &expr, bool value)
   if(expr.id() == ID_equal && value)
   {
     const equal_exprt &equal_expr=to_equal_expr(expr);
+    if(equal_expr.lhs().type().id() == ID_empty)
+    {
+      // ignore equality checking on pointers with empty type
```

Comment:
This comment isn't accurate: I'd suggest "ignore equality checking over expressions with empty (void) type"

Reply:
Thanks, fixed.